src/query/interpreter.cpp

```diff
@@ -1037,21 +1042,28 @@ std::optional<plan::ProfilingStatsWithTotalTime> PullPlan::Pull(AnyStream *strea
   // single `Pull`.
   static constexpr size_t stack_size = 256UL * 1024UL;
   char stack_data[stack_size];
+
   utils::ResourceWithOutOfMemoryException resource_with_exception;
-  utils::MonotonicBufferResource monotonic_memory(&stack_data[0], stack_size, &resource_with_exception);
-  // We can throw on every query because a simple queries for deleting will use only
-  // the stack allocated buffer.
-  // Also, we want to throw only when the query engine requests more memory and not the storage
-  // so we add the exception to the allocator.
-  // TODO (mferencevic): Tune the parameters accordingly.
-  utils::PoolResource pool_memory(128, 1024, &monotonic_memory, utils::NewDeleteResource());
-  std::optional<utils::LimitedMemoryResource> maybe_limited_resource;
+  utils::MonotonicBufferResource monotonic_memory{&stack_data[0], stack_size, &resource_with_exception};
+  std::optional<utils::PoolResource> pool_memory;
+
+  if (use_monotonic_memory_) {
+    pool_memory.emplace(1, kExecutionPoolMaxBlockSize, utils::NewDeleteResource(), utils::NewDeleteResource());
```

Comment:
The idea for `LOAD CSV` is to reuse memory as much as possible, and not to hold or take more than 1 row needs. When the row is processed and the next row comes into place, it would be ideal if we only reuse memory that was already allocated.

kExecutionPoolMaxBlockSize is the size that should guarantee that bigger chunks (bigger lines) should fit inside memory (up to 8000 characters if I am not wrong). And in such cases, it will reuse `Chunk` objects inside `Pool`. Otherwise, it will use unpooled memory. 

To answer the question, setting `blocks_per_chunk`=1 should help to reuse memory more frequently on LOAD CSV and not to allocate more than we need. Idea is that columns inside LOAD  CSV will be of similar size and that we can on for each row reuse memory that we allocated for the row before. If I am not wrong when block_per_chunk is set to 1, for each chunk we have only 1 block inside Chunk, and it is the power of 2 of size, it will only allocate 1 block, as that is how much we will need for that row.

 Don't know if it makes sense to you.

Reply:
@antaljanosbenjamin I made a mistake and my math was a bit of. `kExecutionPoolMaxBlockSize` was intended to be set to 2^15 which is 32 768 bytes as it would allow us to store 4K characters. I checked and for 100 characters we need 800 bytes for the TypedValue vector. 

I agree it is a bit aggressive. My idea and this point is to allocate only what we need. I tested with different sizes 1, 4, and 8 and for LOAD CSV there was no difference in the performance of the query. It seems like a cool idea to me because if there is a need to allocate more, it will happen, but once it does, MemoryResource can reuse it later.

This memory resource is used in EvalContext and since we are dealing with LOAD CSV it should be able to reuse memory, and not suffer performance issues on allocations.